gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStoreTest.java

```diff
@@ -215,6 +246,60 @@ public void testGetSpec() throws Exception {
     Assert.assertEquals(specs.size(), 1);
     Assert.assertTrue(specs.contains(this.flowSpec4));
   }
+  @Test
+  public void testGetSpecsAfterUpdate() throws Exception {
+    this.specStore.addSpec(this.flowSpec5);
+    FlowSpec result = (FlowSpec) this.specStore.getSpec(this.uri5);
+    removeModificationTimestampFromSpecs(result);
+    Assert.assertEquals(result, this.flowSpec5);
+    // Updating  flowSpec5
+    this.specStore.addSpec(this.flowSpec5Updated);
+
+    FlowSpecSearchObject flowSpecSearchObject = FlowSpecSearchObject.builder().flowGroup("fg5").build();
+    Collection<Spec>  specs = this.specStore.getSpecs(flowSpecSearchObject);
+    Assert.assertTrue(specs.contains(this.flowSpec5Updated));
+
+    flowSpecSearchObject = FlowSpecSearchObject.builder().flowName("fn5").build();
+    specs = this.specStore.getSpecs(flowSpecSearchObject);
+    Assert.assertTrue(specs.contains(this.flowSpec5Updated));
+
+    flowSpecSearchObject = FlowSpecSearchObject.builder().flowName("fn5").flowGroup("fg5").build();
+    specs = this.specStore.getSpecs(flowSpecSearchObject);
+    Assert.assertEquals(specs.size(), 1);
+    Assert.assertTrue(specs.contains(this.flowSpec5Updated));
+
+
+    flowSpecSearchObject = FlowSpecSearchObject.builder().propertyFilter(FLOW_SOURCE_IDENTIFIER_KEY+"="+"sourceUpdated").build();
```

Comment:
use the specific methods: e.g. the `FSSO.builder()` has a `.sourceIdentifier(String)` setter for this purpose.  (see the example using `owningGroup(String)` in the test just above.)

unit tests serve double-duty as a system spec, so it's important to show (and exercise!) the API most directly.

in addition those methods more easily enable combining multiple constraints (without delving into the CSV of `propertyFilter`).  sprinkling in an occasional combo search seems preferable to each one in isolation.

overall, do test every facet for which you fixed the "reindexing", in order to catch any error of omission.  (but no need to do one at a time)

Reply:
addressed